shared/src/state-helpers/create-vehicle-parameters.ts

```diff
@@ -11,6 +14,12 @@ import { arrayToUUIDSet } from '../utils/array-to-uuid-set';
 // Be aware that `uuid()` is nondeterministic and cannot be used in a reducer function.
 export function createVehicleParameters(
     vehicleTemplate: VehicleTemplate,
+    materialTemplates: {
+        [key in MaterialType]: MaterialTemplate;
```

Comment:
Dust

Types begin per convention always with a capital letter.

```suggestion
        [Key in MaterialType]: MaterialTemplate;
```

(same for below)